ddtrace/internal/telemetry/metrics_namespaces.py

```diff
@@ -53,22 +34,23 @@ def flush(self):
             }
             return namespace_metrics
 
-    def add_metric(self, metric_type, namespace, name, value=1.0, tags=None, interval=None):
-        # type: (str, str, str, float, MetricTagType, Optional[float]) -> None
+    def add_metric(self, metric_class, namespace, name, value=1.0, tags=None, interval=None):
+        # type: (Type[Metric], str, str, float, MetricTagType, Optional[float]) -> None
         """
         Telemetry Metrics are stored in DD dashboards, check the metrics in datadoghq.com/metric/explorer.
         The metric will store in dashboard as "dd.instrumentation_telemetry_data." + namespace + "." + name
         """
-        metric_id = Metric.get_id(name, namespace, tags, metric_type)
-        metrics_type_payload = self.metrics_type_payloads[metric_type]
+        metric_id = Metric.get_id(name, namespace, tags, metric_class.metric_type)
+        if metric_class is DistributionMetric:
+            metrics_type_payload = TELEMETRY_TYPE_DISTRIBUTION
+        else:
+            metrics_type_payload = TELEMETRY_TYPE_GENERATE_METRICS
```

Comment:
Do we ever expect there to be a different metric class that doesn't follow this rule? Just out of curiosity, non-blocker.

Reply:
Nope. Distributions are their own thing, everything else should be in the general metric payload. There are the main metric formats supported by Datadog. I don't expect this to change anytime soon.